botstrap.py

```diff
@@ -63,66 +79,123 @@ def __getitem__(self, item: str):
 class DiscordClient(Client):
     """An HTTP client to communicate with Discord's APIs."""
 
+    CDN_BASE_URL: Final[str] = "https://cdn.discordapp.com"
+
     def __init__(self, guild_id: int | str):
         super().__init__(
             base_url="https://discord.com/api/v10",
             headers={"Authorization": f"Bot {BOT_TOKEN}"},
             event_hooks={"response": [self._raise_for_status]},
         )
         self.guild_id = guild_id
+        self._app_info: dict[str, Any] | None = None
+        self._guild_info: dict[str, Any] | None = None
+        self._guild_channels: list[dict[str, Any]] | None = None
 
     @staticmethod
     def _raise_for_status(response: Response) -> None:
         response.raise_for_status()
 
+    @property
+    def guild_info(self) -> dict[str, Any]:
+        """Fetches the guild's information."""
+        if self._guild_info is None:
+            response = self.get(f"/guilds/{self.guild_id}")
+            self._guild_info = cast("dict[str, Any]", response.json())
+        return self._guild_info
+
+    @property
+    def guild_channels(self) -> list[dict[str, Any]]:
+        """Fetches the guild's channels."""
+        if self._guild_channels is None:
+            response = self.get(f"/guilds/{self.guild_id}/channels")
+            self._guild_channels = cast("list[dict[str, Any]]", response.json())
+        return self._guild_channels
+
+    def get_channel(self, id: int | str) -> dict[str, Any]:
+        """Fetches a channel by its ID."""
+        for channel in self.guild_channels:
+            if channel["id"] == str(id):
+                return channel
+        raise KeyError(f"Channel with ID {id} not found.")
+    @property
+    def app_info(self) -> dict[str, Any]:
+        """Fetches the application's information."""
+        if self._app_info is None:
+            response = self.get("/applications/@me")
+            self._app_info = cast("dict[str, Any]", response.json())
+        return self._app_info
+
+    def upgrade_application_flags_if_necessary(self) -> bool:
+        """
+        Set the app's flags to allow the intents that we need.
+
+        Returns a boolean defining whether changes were made.
+        """
+        # Fetch first to modify, not overwrite
+        current_flags = self.app_info.get("flags", 0)
+        new_flags = current_flags | 1 << 15 | 1 << 19
+
+        if new_flags != current_flags:
+            resp = self.patch("/applications/@me", json={"flags": new_flags})
+            self._app_info = cast("dict[str, Any]", resp.json())
+            return True
+
+        return False
+
+    def check_if_in_guild(self) -> bool:
+        """Check if the bot is a member of the guild."""
+        try:
+            _ = self.guild_info
+        except HTTPStatusError:
```

Comment:
Maybe we can also check the status here? Avoids us having problems from a transient Discord error that we should surface in a different way.